int64 num
string str